tests/infra/network.py

```diff
@@ -762,17 +762,71 @@ def wait_for_new_primary(self, old_primary_id, timeout_multiplier=2):
         flush_info(logs, None)
         raise error(f"A new primary was not elected after {timeout} seconds")
 
-    def wait_for_snapshot_committed_for(self, seqno, timeout=3):
-        primary, _ = self.find_primary()
+    def wait_for_commit_proof(self, node, seqno, timeout=3):
```

Comment:
Changes in this file are a little awkward but they help us to guarantee that historical queries after a node has joined from a snapshot work. In such scenario, we really want the new node to join from the latest available snapshot. If it didn't verifying the historical query would be trivial as the node would simply have received the historical ledger through classic catch up.

Reply:
This should hopefully get simpler once snapshots can be trivially verified from a receipt over their evidence.